[PPP-3584] PRD - Security Vulnerability when using XML file as datasource

AliaksandrShuhayeu opened the pull request:
[PPP-3584] PRD - Security Vulnerability when using XML file as datasource

- disable default support DTD in pentaho report designer
- add property for enable/disable support DTD

@pamval

pamval commented:
please wait for 7.0 to be branched before merging this

wingman-pentaho commented:
### Build Completed

:fire: This pull request has some issues. It would be preferable to fix them in order for it to be just perfect. See below for more details. Some links are also available below for further assistance in addressing those issues.

#### Build Commands

```sh
ant -Dtestreports.dir=bin/reports/unit-test -f designer/report-designer-assembly/build.xml clean-all resolve jacoco && ant -Dtestreports.dir=bin/reports/integration-test -f designer/report-designer-assembly/build.xml jacoco-integration checkstyle publish-local
ant -Dtestreports.dir=bin/reports/unit-test -f engine/core/build.xml clean-all resolve jacoco && ant -Dtestreports.dir=bin/reports/integration-test -f engine/core/build.xml jacoco-integration checkstyle publish-local
ant -Dtestreports.dir=bin/reports/unit-test -f engine/extensions-xpath/build.xml clean-all resolve jacoco && ant -Dtestreports.dir=bin/reports/integration-test -f engine/extensions-xpath/build.xml jacoco-integration checkstyle publish-local
```

#### Cleanup Commands
```sh
rm -r ~/.ivy2/local || echo no publish local to remove
```

#### Changed files

```console
  designer/report-designer-assembly/resource/resources/classic-engine.properties
  engine/core/source/org/pentaho/reporting/engine/classic/core/classic-engine.properties
  engine/extensions-xpath/source/org/pentaho/reporting/engine/classic/extensions/datasources/xpath/LegacyXPathTableModel.java
```

#### Unit test coverage change

These statistics help you identify how your changes have affected the coverage of the following files. If a file is not in this list, then its coverage was not affected by your changes. To get some help interpreting these metrics, please refer to [Jacoco's documentation](http://eclemma.org/jacoco/trunk/doc/counters.html).

###### org.pentaho.reporting.engine.classic.extensions.datasources.xpath.LegacyXPathTableModel
 * Branch Change:  + .06%
 * Complexity Change: **-1.15%**:small_red_triangle_down:
 * Instruction Change:  + .58%
 * Line Change:  + 1.47%

#### Integration test coverage change

These statistics help you identify how your changes have affected the coverage of the following files. If a file is not in this list, then its coverage was not affected by your changes. To get some help interpreting these metrics, please refer to [Jacoco's documentation](http://eclemma.org/jacoco/trunk/doc/counters.html).

###### org.pentaho.reporting.engine.classic.core.layout.output.DefaultPageStateList.MasterList
 * Branch Change:  + 58.33%
 * Complexity Change:  + 30.00%
 * Instruction Change:  + 55.63%
 * Line Change:  + 50.00%
 * Method Change:  + 50.00%

###### org.pentaho.reporting.engine.classic.core.modules.output.fast.template.DynamicStyleKey
 * Branch Change: **-7.14%**:small_red_triangle_down:
 * Complexity Change: **-9.09%**:small_red_triangle_down:
 * Instruction Change: **-2.22%**:small_red_triangle_down:
 * Line Change: **-4.35%**:small_red_triangle_down:

###### org.pentaho.reporting.engine.classic.core.modules.output.fast.template.DynamicStyleKey.StyleInfoCollection
 * Branch Change: **-6.25%**:small_red_triangle_down:
 * Complexity Change: **-7.14%**:small_red_triangle_down:
 * Instruction Change: **-2.11%**:small_red_triangle_down:
 * Line Change: **-4.17%**:small_red_triangle_down:

###### org.pentaho.reporting.engine.classic.core.util.WeakReferenceList
 * Branch Change:  + 5.00%
 * Complexity Change:  + 8.00%
 * Instruction Change:  + 13.57%
 * Line Change:  + 12.00%
 * Method Change:  + 13.33%

AliaksandrShuhayeu commented:
@pamval @mbatchelor 
I refactored my changes and added tests. 
For full correct testing LegacyXPathTableModel we need deep refactor this calss. 
Now it is not OOP it has actually one method (constructor) with full logic in it.
